test: add the wordpress plugin check action

MaferMazu opened the pull request:
<!--
Please give your pull request a short but descriptive title.
Use conventional commits to separate and summarize commits logically:
https://www.conventionalcommits.org/en/v1.0.0/

Use this template as a guide. Omit sections that don't apply.

🙈 Don't be lazy. Try to fill out the template well.
-->

## Description

This PR adds a GitHub workflow to run [WordPress/plugin-check-action](https://github.com/WordPress/plugin-check-action).

This PR also refactors the workflows.
The zip version generated with this PR is: [openedx-commerce.zip](https://github.com/user-attachments/files/16897592/openedx-commerce.zip)


## Testing instructions

The test `WordPress Plugin Check / Plugin Check` passed.

## Additional information

This test mounts a WordPress environment, installs our plugin using a version similar to our ZIP release, activates the plugin in this new environment, installs the [Plugin check (PCP)](https://wordpress.org/plugins/plugin-check/) by WordPress Performance Team and Plugin Review Team, and run the test over our plugin.

Checks tested: `i18n_usage`, `code_obfuscation`, `direct_db_queries`, `enqueued_scripts_in_footer`, `enqueued_scripts_size`, `enqueued_styles_scope`, `file_type`, `late_escaping`, `localhost`, `no_unfiltered_uploads`, `performant_wp_query_params`, `plugin_header_text_domain`, `plugin_readme`, `plugin_review_phpcs`, `plugin_updater`, `trademarks`.

Categories tested: `accessibility`, `general`, `performance`, `plugin_repo`, `security`.

More info in: [WordPress/plugin-check-action](https://github.com/WordPress/plugin-check-action)

> [!IMPORTANT]  
> This will increase the test execution time by ~ 2 minutes.

## Checklist for Merge

- [x] Tested in a remote environment
- [ ] Updated documentation
- [x] Rebased master/main
- [x] Squashed commits

<!--
You can put NA in the checklist if it doesn't apply

- [x] Check that dont't apply / NA
-->

openedx-webhooks commented:
Thanks for the pull request, @MaferMazu!

### What's next?

*Please work through the following steps to get your changes ready for engineering review:*

#### :radio_button: Get product approval

If you haven't already, [check this list](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3875962884/How+to+submit+an+open+source+contribution+for+Product+Review#Does-my-contribution-require-Product-Review%3F) to see if your contribution needs to go through the product review process.

- If it does, you'll need to submit a product proposal for your contribution, and have it reviewed by the [Product Working Group](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3449028609/Product+Working+Group).
    - This process (including the steps you'll need to take) is documented [here](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3875962884/How+to+submit+an+open+source+contribution+for+Product+Review#Product-Review-Process).
- If it doesn't, simply proceed with the next step.

#### :radio_button: Provide context

To help your reviewers and other members of the community understand the purpose and larger context of your changes, feel free to add as much of the following information to the PR description as you can:

- Dependencies
  > This PR must be merged before / after / at the same time as ...
- Blockers
  > This PR is waiting for OEP-1234 to be accepted.
- Timeline information
  > This PR must be merged by XX date **because** ...
- Partner information
  > This is for a course on edx.org.
- Supporting documentation
- Relevant [Open edX discussion forum](https://discuss.openedx.org/) threads





#### :radio_button: Get a green build

If one or more checks are failing, continue working on your changes until this is no longer the case and your build turns green.


#### :radio_button: Update the status of your PR

Your PR is currently marked as a draft.  After completing the steps above, update its status by clicking "Ready for Review", or removing "WIP" from the title, as appropriate.

<!-- comment:end_of_wip -->


#### :radio_button: Let us know that your PR is ready for review:

### Who will review my changes?


This repository is currently maintained by `@mafermazu`.  Tag them in a comment and let them know that your changes are ready for review.


### Where can I find more information?

If you'd like to get more details on all aspects of the review process for open source pull requests (OSPRs), check out the following resources:

- [Overview of Review Process for Community Contributions](https://docs.openedx.org/en/latest/developers/references/developer_guide/process/FAQ-about-pull-requests.html)
- [Pull Request Status Guide](https://docs.openedx.org/en/latest/developers/references/developer_guide/process/pull-request-statuses.html)
- [Making changes to your pull request](https://docs.openedx.org/en/latest/documentors/how-tos/make_changes_to_your_pull_request.html)

### When can I expect my changes to be merged?

Our goal is to get community contributions seen and reviewed as efficiently as possible.

However, the amount of time that it takes to review and merge a PR can vary significantly based on factors such as:

- The size and impact of the changes that it introduces
- The need for product review
- Maintenance status of the parent repository

:bulb: *As a result it may take up to several weeks or months to complete a review and merge your PR.*

<!-- comment:external_pr -->
<!-- data: eyJkcmFmdCI6IHRydWV9 -->

felipemontoya commented:
This looks fantastic.

Have you tried this failing? we can make a new PR against this branch that does one of the things that should break this plugin check.

felipemontoya commented on .github/workflows/wordpress-plugin-check.yml:
```diff
@@ -0,0 +1,27 @@
+    - name: Create openedx-commerce directory and copy files
+      run: |
+        mkdir openedx-commerce
+        rsync -av --exclude='*.git*' --exclude='.*' --exclude='/test/*' --exclude='/requirements/*' --exclude='/docs/*' --exclude='composer.lock' --exclude='*.yaml' --exclude='*.xml' --exclude='Makefile' ./ openedx-commerce/
```
I don't like that we have a duplicate of the process in this line and this [action in a different workflow](https://github.com/openedx/openedx-wordpress-ecommerce/blob/main/.github/workflows/upload-release-zip.yml#L22-L27).

Do you think we could refactor both to use a `make release` target or something like this? What I really want is that we only have one source of truth when it comes to what this action tests and what the release zip finally contains

MaferMazu replied:
I thought about it but wasn't sure where to implement it.

I'll implement that in the Makefile to use it in both. Thanks.

MaferMazu commented:
Here is a PR with `WordPress Plugin Check / Plugin Check` failing: https://github.com/openedx/openedx-wordpress-ecommerce/pull/94
Action failing: https://github.com/openedx/openedx-wordpress-ecommerce/actions/runs/10722515445/job/29733613522?pr=94
Change that makes it fail: https://github.com/openedx/openedx-wordpress-ecommerce/pull/94/commits/3e09dc1f1c12d87eb55428ff4037dc0810a52ac9

MaferMazu commented:
@felipemontoya, thanks for the feedback. I refactored the workflows, and the zip version generated is [openedx-commerce.zip](https://github.com/user-attachments/files/16897592/openedx-commerce.zip)

Note: I added the cd command in the zip process to zip the content but not the folder.

This is ready for a re-review.

felipemontoya reviewed (approved):
This looks perfect. Thanks a lot
